purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -217,44 +219,25 @@ class Purchases internal constructor(
 
     /**
      * This method will send all the purchases to the RevenueCat backend. Call this when using your own implementation
-     * for subscriptions anytime a sync is needed, such as when migrating existing users to RevenueCat
+     * for subscriptions anytime a sync is needed, such as when migrating existing users to RevenueCat. The
+     * [SyncPurchasesCallback.onSuccess] callback will be called if all purchases have been synced successfully or
+     * there are no purchases. Otherwise, the [SyncPurchasesCallback.onError] callback will be called with a
+     * [PurchasesError] indicating the first error found.
      *
+     * @param [listener] Called when all purchases have been synced with the backend, either successfully or with
+     * an error. If no purchases are present, the success function will be called.
      * @warning This function should only be called if you're migrating to RevenueCat or in observer mode.
+     * @warning This function could take a relatively long time to execute, depending on the amount of purchases
+     * the user has. Consider that when waiting for this operation to complete.
```

Comment:
I added this new warning to the documentation so developers consider whether they need to wait for this to continue when using this function.